lib/Controller/WopiController.php

```diff
@@ -562,7 +562,8 @@ public function putFile($fileId,
 				$suggested = iconv('utf-7', 'utf-8', $suggested);
 
 				if ($suggested[0] === '.') {
-					$path = dirname($file->getPath()) . '/New File' . $suggested;
+					$rawName = pathinfo($file->getName(), PATHINFO_FILENAME);
```

Comment:
If i understand correctly, `X-WOPI-SuggestedTarget` provides either a file name or a extension. Now in this case, when the provided string starts with a '.', pathinfo will always provide an empty string for `PATHINFO_FILENAME`, according to the documentation.

Unless a path is provided with this header, which is not what I understand it transports, but i would not exclude it entirely… but then a check would be missing.

https://3v4l.org/#live